portfolio/src/main/webapp/gallery.js

```diff
@@ -3,55 +3,57 @@ let getLatestOpenedImg;
-        "/images/photography/thumbnails/"
-      );
-      //get second part of the split URL and remove the ")
-      let setNewImageURL = getImageURLPos[1].replace('")', "");
+function init() {
+    //check if there are gallery images
+    if (galleryImages) {
+        //for each image and take in index of image in the array
+        galleryImages.forEach(function(image, index) {
+            //get full resolution image
+            image.onclick = function() {
+            //get css style of selected image
+            let getElementsCss = window.getComputedStyle(image);
+            //get background image property of the image from css style list
+            let getFullImageURL = getElementsCss.getPropertyValue("background-image");
+            //get last part of background image url, the image name
+            let getImageURLPos = getFullImageURL.split(
+                "/images/photography/thumbnails/"
+            );
+            //get second part of the split URL and remove the ")
+            let setNewImageURL = getImageURLPos[1].replace('")', "");
 
-      getLatestOpenedImg = index + 1;
+            getLatestOpenedImg = index + 1;
 
-      let container = document.body;
-      let newImageWindow = document.createElement("div");
-      //apply div to body of website
-      container.appendChild(newImageWindow);
-      newImageWindow.setAttribute("class", "img-window");
-      newImageWindow.setAttribute("onclick", "closeImg()");
+            let container = document.body;
+            let newImageWindow = document.createElement("div");
+            //apply div to body of website
+            container.appendChild(newImageWindow);
+            newImageWindow.setAttribute("class", "img-window");
+            newImageWindow.setAttribute("onclick", "closeImg()");
 
-      let newImage = document.createElement("img");
-      newImageWindow.appendChild(newImage);
-      newImage.setAttribute("src", "images/photography/" + setNewImageURL);
-      newImage.setAttribute("id", "current-img");
+            let newImage = document.createElement("img");
+            newImageWindow.appendChild(newImage);
+            newImage.setAttribute("src", "images/photography/" + setNewImageURL);
+            newImage.setAttribute("id", "current-img");
 
-      //execute code after image is loaded
-      newImage.onload = function() {
-        let newNextBtn = document.createElement("img");
-        container.appendChild(newNextBtn);
-        newNextBtn.setAttribute("src", "images/right-arrow.png");
-        newNextBtn.setAttribute("class", "img-btn-next");
-        newNextBtn.setAttribute("onclick", "changeImg(1)");
-        newNextBtn.style.cssText = "right: 20px";
+            //execute code after image is loaded
+            newImage.onload = function() {
+                let newNextBtn = document.createElement("img");
+                container.appendChild(newNextBtn);
+                newNextBtn.setAttribute("src", "images/right-arrow.png");
```

Comment:
I'm seeing some duplicate logic in gallery.js and art-gallery.js. Could you pull out any duplicate code into a new file that both files import?

Reply:
just pushed again after factoring out code